simulators/ethereum/rpc-compat/main.go

```diff
@@ -177,33 +177,98 @@ func postHttp(c *http.Client, url string, d io.Reader) ([]byte, error) {
-		resp = deleteField(resp, "result.hash")
-		expectedData = deleteField(expectedData, "result.hash")
-	}
+	resp, expectedData = cleanBlockData(resp, expectedData)
+
+	resp, expectedData = cleanReceiptsData(resp, expectedData)
+
+	return resp, expectedData
+}
+
+// 🚧 WARNING KAKAROT
+// cleanBlockData modifies the response and expected data if the returned values match a block information.
+func cleanBlockData(resp, expectedData string) (string, string) {
+	var fields = []string{"result.hash", "result.parentHash", "result.timestamp", "result.baseFeePerGas", "result.difficulty", "result.gasLimit", "result.miner", "result.size", "result.stateRoot", "result.totalDifficulty", "result.withdrawals"}
+	resp = deleteFields(resp, fields...)
+	expectedData = deleteFields(expectedData, fields...)
 
 	if gjson.Get(resp, "result.transactions").Exists() && gjson.Get(expectedData, "result.transactions").Exists() {
 		transactions := gjson.Get(resp, "result.transactions").Array()
 		for i := range transactions {
-			resp, _ = sjson.Delete(resp, fmt.Sprintf("result.transactions.%d.blockHash", i))
+			resp = deleteField(resp, fmt.Sprintf("result.transactions.%d.blockHash", i))
 		}
 
 		transactions = gjson.Get(expectedData, "result.transactions").Array()
 		for i := range transactions {
-			expectedData, _ = sjson.Delete(resp, fmt.Sprintf("result.transactions.%d.blockHash", i))
+			expectedData = deleteField(expectedData, fmt.Sprintf("result.transactions.%d.blockHash", i))
+		}
+	}
+
+	return resp, expectedData
+}
+
+// 🚧 WARNING KAKAROT
+// cleanReceiptsData modifies the response and expected data if the returned values match a []receipt.
+func cleanReceiptsData(resp, expectedData string) (string, string) {
+	receipts := gjson.Get(resp, "result").Array()
+	for i := range receipts {
+		var fields = []string{fmt.Sprintf("result.%d.blockHash", i), fmt.Sprintf("result.%d.cumulativeGasUsed", i), fmt.Sprintf("result.%d.gasUsed", i)}
+		resp = deleteFields(resp, fields...)
+	}
+
+	receipts = gjson.Get(expectedData, "result").Array()
+	for i := range receipts {
+		var fields = []string{fmt.Sprintf("result.%d.blockHash", i), fmt.Sprintf("result.%d.cumulativeGasUsed", i), fmt.Sprintf("result.%d.gasUsed", i)}
+		expectedData = deleteFields(expectedData, fields...)
+	}
+
+	return resp, expectedData
+}
+
+// 🚧 WARNING KAKAROT
+// cleanTransactionData modifies the response and expected data if the returned values match a receipt.
+func cleanTransactionData(resp, expectedData string) (string, string) {
+	resp = deleteField(resp, "result.blockHash")
+	expectedData = deleteField(expectedData, "result.blockHash")
+
+	return resp, expectedData
+}
+
+// 🚧 WARNING KAKAROT
+// cleanReceiptData modifies the response and expected data if the returned values match a receipt.
+func cleanReceiptData(resp, expectedData string) (string, string) {
+	resp = deleteFields(resp, "result.gasUsed", "result.cumulativeGasUsed")
```

Comment:
should we "skip" gasUsed and cumulativeGasUsed?

since eni will fix it soon?

or add a TODO to add it back when gas accounting is precise?

Reply:
Will add a issue!

Reply:
ahhh we can't because it's a fork, ok will add a TODO